group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -842,33 +838,40 @@ ConsumerGroup consumerGroup(
      *                          created if it does not exist.
      *
      * @return A ConsumerGroup.
-     * @throws IllegalStateException if the group does not exist and createIfNotExists is false or
-     *                               if the group is not a consumer group.
+     * @throws GroupIdNotFoundException if the group does not exist and createIfNotExists is false or
+     *                                  if the group is not a consumer group.
+     * @throws IllegalStateException    if the group does not have the expected type.
      * Package private for testing.
      */
     ConsumerGroup getOrMaybeCreatePersistedConsumerGroup(
         String groupId,
         boolean createIfNotExists
-    ) throws GroupIdNotFoundException {
+    ) throws IllegalStateException {
```

Comment:
Silly question, but since both exceptions are in this method, why do we choose one or the other in the signature?

Reply:
My bad. Both should be there.

Reply:
I can fix tomorrow or as a follow up.